sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -115,43 +101,41 @@ class SimpleCatalog(val caseSensitive: Boolean) extends Catalog {
 trait OverrideCatalog extends Catalog {
 
   // TODO: This doesn't work when the database changes...
-  val overrides = new mutable.HashMap[(Option[String],String), LogicalPlan]()
+  val overrides = new mutable.HashMap[String, LogicalPlan]()
```

Comment:
Why collapse this to a `String`?  What about a table with a name like the following:
```
SELECT * FROM `database`.`table.name`
```